dsa-23au/java-dsa/arrays-links/src/main/java/dev/codewithfriends/Stack.java

```diff
@@ -0,0 +1,36 @@
+package dev.codewithfriends
+
+//2 methods
+//push pop :)
+//use nodes
+
+public class stack<T>
+{
+    private T item;
+    private Node<T> top;
+
+   public void push(T item)
+   {
+
+    private Node <T> newNode = new Node<> (item);
+    newNode.Next = top;
+    top = newNode;
+    
+   }
+
+   public <T> pop(T item)
+   {
+
+    if(isEmpty()) 
+    {
+        return null;
+        T item = top.item;
+        top = top.Next;
+        return item;
+    }
+
```

Comment:
[Assignment and Co-Creation Week 04](https://github.com/TheEvergreenStateCollege/upper-division-cs/blob/main/dsa-23au/notes/docs/week-04/Assignment-04.md#assignment-and-co-creation-week-04)

Inciteful comment